### PR TITLE
Get Employee By UserName

### PR DESCRIPTION
**Pull request by mudit1102**

### CODE REVIEW

File: src/main/java/com/work/management/service/employee/EmployeeService.java [hunk 1]
```diff
@@ -1,6 +1,10 @@
 package com.work.management.service.employee;
 
+import com.work.management.dto.EmployeeDto;
+
 public interface EmployeeService {
 
   void save(com.work.management.dto.EmployeeDto document);
+
```
Comment (ankitmathur3193) [inline src/main/java/com/work/management/service/employee/EmployeeService.java:8]: remove com.work.management.dto

File: src/main/java/com/work/management/web/rest/controller/EmployeeController.java [hunk 2]
```diff
@@ -32,4 +33,14 @@
...
     return new ResponseEntity<>(employeeResource, HttpStatus.CREATED);
   }
+
+  @RequestMapping(method = RequestMethod.GET, value = "/{userName}", produces = "application/json; charset=UTF-8")
+  @ApiOperation(value = "Get an employee by its user name", response = EmployeeResource.class)
+  ResponseEntity<EmployeeResource> getEmployeeByUserName(
+      @Valid @ApiParam(value = "User Name", required = true) @PathVariable("userName") String userName) {
+
+    EmployeeResource employeeResource = EmployeeAssembler
+        .convert(employeeService.getEmployeeByUserName(userName));
```
Comment (ankitmathur3193) [inline src/main/java/com/work/management/web/rest/controller/EmployeeController.java]: Inline this variable(employeeResource)...cltr+shift+i...as your are not using it anywhere else...and remove the empty line at 41.

File: src/main/java/com/work/management/service/employee/EmployeeService.java [hunk 3]
```diff
@@ -1,6 +1,10 @@
 package com.work.management.service.employee;
 
+import com.work.management.dto.EmployeeDto;
+
 public interface EmployeeService {
 
-  void save(com.work.management.dto.EmployeeDto document);
+  void save(EmployeeDto document);
```
Comment (ankitmathur3193) [inline src/main/java/com/work/management/service/employee/EmployeeService.java]: change name from document to employeeDto

**Review by ankitmathur3193 — DISMISSED** [review]

**Comment by mudit1102** [comment]
Done with review comments

**Pull request merged**